libs/client-api/src/v2/db.rs

```diff
@@ -129,24 +131,31 @@ impl Db {
     Ok(())
   }
 
+  #[instrument(level = "trace", skip_all, err)]
   pub fn save_update(
     &self,
     object_id: &ObjectId,
     message_id: Option<Rid>,
     update_v1: &[u8],
+    action_source: ActionSource,
   ) -> Result<Option<StateVector>, PersistenceError> {
     if update_v1 == Update::EMPTY_V1 {
       tracing::trace!("skipping empty update {}", object_id);
       return Ok(None);
     }
     let instance = self.inner.get()?;
     let ops = instance.write_txn();
+
+    #[cfg(feature = "verbose_log")]
     tracing::trace!(
-      "persisting update for {}/{} by {}",
+      "persisting {} update for {}/{} by {}. update: {:#?}",
+      action_source,
```

Comment:
If the only reason for `action_source` to be passed as a parameter is to be logged inside of the method, maybe it's better to put the log line before this method is called?